### DISCUSSION TITLE
Converting raw BGR image to PNG writes problematic gamma value into file

### DISCUSSION THREAD

**Discussion by dreamlayers** [ImageMagick/ImageMagick#4752] [Help]
I was converting raw BGR images to PNG using `convert -depth 8 -size 2592x1944 infile.bgr outfile.png` using ImageMagick 6.9.11-60 Q16 x86_64 in Ubuntu 21.10. The resulting PNG files look very washed out in Gwenview but fine in Geeqie and some other software.  If I strip out gamma and colour information using `optipng -o0 -strip all  i.png` then the image displays properly in Gwenview.  Either before or after stripping of data via optipng, I can `convert outfile.png checkfile.bgr` and get a file that's identical to infile.bgr.  

I can affect the appearance of the image in Gwenview by setting gamma when converting. Using `convert -depth 8 -size 2592x1944 infile.bgr -set gamma 10 outfile.png` I get a very dark image instead of a washed out one. If I convert it back to bgr, it is not identical to the original, though if I first strip data via optipng, it will be identical to the original. So, the gamma value in the file seems to be what makes the image look different in Gwenview. 

The best thing I can do here seems to be stripping the unnecessary extra data from PNG files via optipng. But it would be nice to not have to do that.

**Comment by snibgo** [comment]
Do you know the colorspace of your input BGR? IM assumes it is sRGB, but perhaps it isn't really.

When I convert a sRGB PNG image to BGR, and convert that back to PNG, I get the same pixel values, the same metadata as far as I can see, and the image looks the same in Windows Photo Viewer.

Perhaps you can link to your input BGR.
